src/main/java/nl/knaw/dans/sword2/core/service/ZipServiceImpl.java

```diff
@@ -49,7 +50,7 @@ void extractZipFileWithFileMapping(ZipFile zipFile, Path targetPath, Map<String,
 
         zipFile.stream().filter(e -> !e.getName().endsWith("/")).forEach(entry -> {
             var name = entry.getName();
-            if (name.contains("../")) {
+            if (!targetPath.resolve(name).normalize().startsWith(targetPath.normalize())) {
```

Comment:
@ericdevries : latest of a number of attempts to fix this properly. The previous ones I merged without PR. It would probably be better to reject the deposit instead of ignoring the rogue entry. Not sure how to do that within the stream prcocessing though. Any ideas?